src/components/GroupsTable/GroupsTable.js

```diff
@@ -0,0 +1,237 @@
+    };
+
+    const filterConfigItems = [
+        {
+            type: 'custom',
+            label: 'Name',
+            filterValues: {
+                children: (
+                    <SearchInput
+                        data-ouia-component-type="PF4/TextInput"
+                        data-ouia-component-id="name-filter"
+                        placeholder="Filter by name"
+                        value={filters.hostname_or_id || ''}
+                        onChange={(value) => {
+                            const { hostname_or_id, ...fs } = filters;
+                            return setFilters({
+                                ...fs,
+                                ...(value.length > 0 ? { hostname_or_id: value } : {})
+                            });
+                        }}
+                        onClear={() => {
+                            const { hostname_or_id, ...fs } = filters;
+                            return setFilters(fs);
+                        }}
+                        isDisabled={rejected}
+                    />
+                )
+            }
+        }
+    ];
+
+    const onResetFilters = () => setFilters(GROUPS_TABLE_INITIAL_STATE);
+
+    const activeFiltersConfig = {
+        showDeleteButton: !!filters.hostname_or_id,
+        deleteTitle: 'Reset filters',
+        filters: filters.hostname_or_id
+            ? [
+                {
+                    category: 'Name',
+                    chips: [
+                        { name: filters.hostname_or_id, value: filters.hostname_or_id }
+                    ]
+                }
+            ]
+            : [],
+        // always reset to initial filters since there is only one filter currently
+        onDelete: onResetFilters
+    };
+
+    const onSetPage = (event, page) => setFilters({ ...filters, page });
+
+    const onPerPageSelect = (event, perPage) =>
+        setFilters({ ...filters, perPage, page: 1 }); // will also reset the page to first
+
+    // TODO: use ouiaSafe to indicate the loading state for e2e tests
+
+    return (
+        <div id="groups-table">
+            <PrimaryToolbar
+                pagination={{
+                    itemCount: data?.total || 0,
+                    page: filters.page,
+                    perPage: filters.perPage,
+                    onSetPage,
+                    onPerPageSelect,
+                    isCompact: true,
+                    ouiaId: 'pager',
+                    isDisabled: rejected
+                }}
+                filterConfig={{ items: filterConfigItems }}
+                activeFiltersConfig={activeFiltersConfig}
+            />
+            <Table
+                aria-label="Groups table"
+                ouiaId="groups-table"
+                /* ouiaSafe={!loadingState}> */
+                variant={TableVariant.compact}
+                cells={GROUPS_TABLE_COLUMNS}
+                rows={
+                    uninitialized || loading
```

Comment:
I think this can be moved to a helper function to have cleaner code. What do you think?